### PR TITLE
WEB-1160: New form building method

### PR DESCRIPTION
**Pull request by andyparisi**
- **Added new method: `updateFormValue`**. This method behaves essentially the same as `buildFormData`, but does not iterate over `this.refs`. Instead, it parses a `key` parameter on an individual basis, intended for use with `onChange` events. Singular updates are made to JSON structures.

- Added new `jsonPath` prop to our various input components. Will be returned alongside `value` in this format: `onChange(value, jsonPath)`. The `jsonPath` is to be passed into `updateFormValue` as the second parameter (the key).

- Minor tweaks made to documentation generation. Now allowing a more generic `@param` tag in place of `@props`. _Note: Document generation is still limited to one comment block per document._

- Updated test suite description, as the current `buildFormData` tests also cover the same behavior as the new method. These tests will probably be bolstered as we integrate the new method into forms.

### CODE REVIEW

File: src/components/select_input_custom.coffee [hunk 1]
```diff
@@ -260,14 +260,14 @@ SelectInputCustom = createClass
...
-    {onChange, validation} = @props
+    {onChange, validation, jsonPath} = @props
     {valueHasChanged} = @state
 
     @setState {valueHasChanged: yes}, =>
       @validate(validation, value) 
       
       @value = value
-      onChange?(value)
+      onChange?(@getValue(), jsonPath)
```
Comment (darcyadams) [inline src/components/select_input_custom.coffee]: why use `@getValue` here and not `value`? `@getValue` uses refs internally.

Comment (andyparisi) [reply src/components/select_input_custom.coffee]: The reason is that the `getValue` method has a bunch of logic that handles `returnFullObjects`, `textTransform`, and `returnNull`. I didn't want to replicate all this logic. Also, the only string ref that method uses is `@refs.input`, which we can easily refactor to `@input` at some point.

**Comment by darcyadams** [comment]
I think you forgot `multi_select`

**Comment by darcyadams** [comment]
... and `ios_toggle_switch`

**Comment by andyparisi** [comment]
Yeah i was afraid I'd miss some spots... updated

**Pull request merged**